src/qscan.cpp

```diff
@@ -0,0 +1,874 @@
+			printf(IMEDIA "Supplement. SA: %d (%dMB) / %d (%dMB)\n",
+				dev->media.spare_ssa_free,
+				dev->media.spare_ssa_free/512,
+				dev->media.spare_ssa_total,
+				dev->media.spare_ssa_total/512);
+		}
+
+		if ( !(dev->media.type & (DISC_CDROM | DISC_DVDROM)) )
+			printf(IMEDIA "Media ID      : %s\n", dev->media.MID);
+		if ( dev->media.type & (DISC_DVDminus) )
+			printf(IMEDIA "Written on    : %s\n", dev->media.writer);
+		if ((flags & FL_MID_RAW) && (dev->media.type & DISC_CD) && dev->media.ATIP_size) {
+			for (int i=0; i<((dev->media.ATIP_size-4) >> 2); i++) {
+				printf(IMEDIA "ATIP data %X   : ", i);
+					printf(" %02X %02X %02X %02X",
+						dev->media.ATIP[(i<<2)+4], dev->media.ATIP[(i<<2)+5], dev->media.ATIP[(i<<2)+6], dev->media.ATIP[(i<<2)+7]);
+					printf(" | %3d %3d %3d %3d\n",
+						dev->media.ATIP[(i<<2)+4], dev->media.ATIP[(i<<2)+5], dev->media.ATIP[(i<<2)+6], dev->media.ATIP[(i<<2)+7]);
+			}
+		}
+		if (dev->media.MID_size) {
+			if (dev->parms.wr_speed_tbl_media[0] > 0) {
+				printf(SMEDIA "Manufacturer defined write speeds:\n");
+				for (int i=0; i<speed_tbl_size && dev->parms.wr_speed_tbl_media[i] > 0; i++)
+				//for (int i=0; i<speed_tbl_size; i++)
+					printf(SMEDIA "M WR speed #%02d: %4.1fX\n", i, dev->parms.wr_speed_tbl_media[i]);
+			}
+			if (flags & FL_MID_RAW) {
+				for (int i=0; i<(dev->media.MID_size >> 4); i++) {
+					unsigned char c;
+					printf(IMEDIA "MID raw data %X0: ", i);
+					for (int ii=0; ii<16 && ( (i<<4) +ii) < dev->media.MID_size ; ii++) {
+						printf("%02X ", (unsigned char) dev->media.MID_raw[(i<<4) + ii + 4]);
+					}
+					printf(" | ");
+					for (int ii=0; ii<16 && ( (i<<4) +ii) < dev->media.MID_size ; ii++) {
+						c = dev->media.MID_raw[(i<<4) + ii + 4];
+						printf("%c", (c > 32) ? c : '.' );
+					}
+					printf("\n");
+				}
+			}
+		}
+		get_track_list(dev);
+		printf(IMEDIA "Total tracks  : %d\n", dev->media.tracks);
+		printf(IMEDIA "Track:     Session   TMode   DMode      Start        End       Size       Free   Next Writable\n");
+		printf("--------------------------------------------------------------------------------------------------\n");
+		if (dev->media.tracks) for (i=0; i<dev->media.tracks; i++) {
+			printf(IMEDIA "Track# %2d       %2d      %2d      %2d   %8d   %8d   %8d   %8d   %8d\n",
+				dev->media.track[i].n,
+				dev->media.track[i].session,
+				dev->media.track[i].track_mode,
+				dev->media.track[i].data_mode,
+				dev->media.track[i].start,
+				dev->media.track[i].last,
+				dev->media.track[i].size,
+				dev->media.track[i].free,
+				dev->media.track[i].next_writable);
+		}
+		printf("--------------------------------------------------------------------------------------------------\n\n");	
+		goto end;
+	}
+	if (test && !strcmp(test,"wt")) {
+		if (!dev->media.type) {
+			printf("No media!\n");
+			result=2;
+			goto end;
+		}
+		if (dev->media.dstatus & !(dev->media.type & (DISC_DVDRAM | DISC_DVDpRW | DISC_DVDpRWDL))) {
```

Comment:
Codacy found an issue: [Boolean result is used in bitwise operation. Clarify expression with parentheses.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)